### PR TITLE
[terra-clinical-item-display] Deprecating isDisabled prop

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

Deprecated `isDisabled` prop as this feature does not meet accessibility guidelines. 

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8934 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by sycombs — APPROVED** [review]

**Review by eawww — COMMENTED** [review]
Could you also remove the isDisabled example from the doc site?

**Comment by sdadn** [comment]
> Could you also remove the isDisabled example from the doc site?

Since the prop is still usable, would it make sense to keep the example and remove it when the prop is fully removed?

File: packages/terra-clinical-item-display/src/ItemDisplay.jsx [hunk 1]
```diff
@@ -82,6 +83,11 @@ const ItemDisplay = ({
     { 'strike-through': textStyle === TextStyles.STRIKETHROUGH },
   ]);
 
+  // TODO: remove this warning in the next major release.
+  if (isDisabled) {
```
Comment (smason0) [inline packages/terra-clinical-item-display/src/ItemDisplay.jsx:87]: I see it's just a warning, but should we disable the no-console linter rule for this line?

File: packages/terra-clinical-item-display/src/ItemDisplay.jsx [hunk 2]
```diff
@@ -29,8 +29,9 @@ const propTypes = {
    * Whether or not the text should be truncated.
    */
   isTruncated: PropTypes.bool,
+  //  TODO: remove isDisabled in the next major release.
```
Comment (smason0) [inline packages/terra-clinical-item-display/src/ItemDisplay.jsx:32]: Is there a Jira logged to go back and address these TODOs?

**Comment by sdadn** [comment]
> > Since the prop is still usable, would it make sense to keep the example and remove it when the prop is fully removed?
> 
> @sdadn I wouldn't say so. The example is to instruct consumers on consuming it. Nobody should be implementing what the example shows now that it's deprecated. I can't think of a benefit it provides and may actually encourage new use of it.

Since this doesn't affect passivity, I went ahead and removed the examples here:
d709832991c616adbf63cab791d2a27c91641c72

File: packages/terra-clinical-item-display/src/ItemDisplay.jsx (hunk 1 above)
Comment (sdadn) [reply packages/terra-clinical-item-display/src/ItemDisplay.jsx:87]: Disabled here: 51180274acf452a3d57b7018e52ed217271efdfe

**Review by eawww — APPROVED** [review]

**Review by smason0 — APPROVED** [review]

**Pull request merged**